plutus-ledger-api/test-onchain-evaluation/Main.hs

```diff
@@ -19,46 +19,53 @@ import System.Environment (getEnv)
 import System.FilePath (isExtensionOf)
 import Test.Tasty
 import Test.Tasty.HUnit
+import Test.Tasty.Providers
+
+data SingleTest
+    = SingleTest
+        LedgerPlutusVersion
+        (Maybe (EvaluationContext, [Integer]))
+        ScriptEvaluationEvent
+
+instance IsTest SingleTest where
+    run _opts (SingleTest ver mbCtx ev) _reportProgress = case mbCtx of
+        Just (ctx, params) ->
+            evaluate
+                =<< ( pure . maybe (testPassed mempty) (testFailed . display) $
+                        checkEvaluationEvent ctx params ev
+                    )
```

Comment:
https://github.com/input-output-hk/plutus/blob/master/STYLEGUIDE.adoc#avoid-multi-line-parenthesized-expressions

What's wrong with a good old do block?

Reply:
Nothing :slightly_smiling_face: Whenever a do-block is like `do res <- foo; bar res` I tend to try to avoid `res` variable via `=<<`. No objection to using a do-block.

Reply:
I do the opposite. I like names and structured syntax :)

Reply:
That's fine, but by the way I don't quite agree with that advice in the style guide :slightly_smiling_face: 

It really depends on how you format it. I think

```haskell
fmap
   (
    ...
   )
   xs
```

is quite clear and readable. But

```haskell
fmap (...
   ... ) xs
```

is no good.